code/modules/client/preferences/multiz_performance.dm

```diff
@@ -0,0 +1,25 @@
+/// We expect at most 3 layers of multiz
+/// Increment this define if you make a huge map. this has a layer of wiggle room since I don't trust you all
+/// If you modify this, you'll need to modify the tsx file too
+#define MAX_EXPECTED_Z_DEPTH 3
```

Comment:
`#undef` this at the end, but also you should probably unit test this? Like add a unit test to make sure the map size is never smaller than the number of station levels. We test on every map so it will work

Reply:
uhhhh yeah sure that'll do